client-server-client/src/main/scala/zio/raft/client/ClientConfig.scala

```diff
@@ -0,0 +1,43 @@
+package zio.raft.client
+
+import zio._
+
+/**
+ * Client configuration for ZIO Raft client-server communication.
+ */
+case class ClientConfig(
+  clusterAddresses: List[String],
```

Comment:
at the moment we ignore and hints on who is actually the leader. ClusterAddresses should be a map from MemberId to address, we should then not ignore the hints, and try to connect to the leader whenever we disconnect due to NotALeader